flake.nix

```diff
@@ -1,29 +1,134 @@
+            version = crateNameFromCargoToml.version;
+            src = lib.fileset.toSource {
+              root = ./.;
+              fileset = fs.unions [
+                (fs.fileFilter (file: file.hasExt "sol") ./vendor/solidity)
+                (fs.fileFilter (file: file.hasExt "sol") ./packages/ethereum/contracts/src)
+                ./packages/ethereum/contracts/foundry.toml
+                ./packages/ethereum/contracts/remappings.txt
+              ];
+            };
+            buildInputs = with pkgs; [ foundry-bin solcDefault ];
+            buildPhase = ''
+              cd ./packages/ethereum/contracts
+              sed -i "s|solc = .*|solc = \"${solcDefault}/bin/solc\"|g" foundry.toml
+              forge bind --bindings-path ../crates/bindings --crate-name bindings \
+                --overwrite --select '^Hopr.*?(Boost|[^t])$$'
+            '';
+            installPhase = ''
+              echo `pwd`
+              cp -r --no-preserve=mode,ownership ../crates/bindings $out/
+            '';
+          };
+          src = fs.toSource {
+            root = ./.;
+            fileset = fs.unions [
+              ./vendor/cargo
+              ./.cargo/config.toml
+              ./Cargo.lock
+              (fs.fileFilter (file: file.hasExt "rs") ./.)
+              (fs.fileFilter (file: file.hasExt "toml") ./.)
+              ./packages/hoprd/crates/hopr-lib/data
+            ];
+          };
+          rustToolchain = pkgs.pkgsBuildHost.rust-bin.fromRustupToolchainFile ./rust-toolchain.toml;
+          nativeBuildInputs = with pkgs; [
+            rustToolchain
+            pkg-config
+          ];
+          buildInputs = with pkgs; [
+            foundry-bin
+          ] ++ pkgs.lib.optionals pkgs.stdenv.isDarwin (
+            with darwin.apple_sdk.frameworks; [
+              SystemConfiguration
+            ]
+          );
+          commonArgs = {
+            inherit (crateNameFromCargoToml) pname version;
+            inherit src buildInputs nativeBuildInputs;
+            CARGO_HOME = ".cargo";
+            cargoVendorDir = "./vendor/cargo";
+            # disable running tests automatically for now
+            doCheck = false;
+          };
+          cargoArtifacts = craneLib.buildDepsOnly (commonArgs // {
+            extraDummyScript = ''
+              rm -rf $out/vendor/cargo
+              cp -r --no-preserve=mode,ownership ${src}/vendor/cargo $out/vendor/
+            '';
+          });
+          hoprd = craneLib.buildPackage (commonArgs // {
+            inherit cargoArtifacts;
+            preBuild = ''
+              cp -r ${ethereumBindings}/src ./packages/ethereum/crates/bindings/
+              cp .cargo/config.toml vendor/cargo/
+            '';
+          });
+          # FIXME: the docker image built is not working on macOS arm platforms
+          # and will simply be a no-op. Likely, some form of cross-compilation
+          # or distributed build is required.
```

Comment:
We want a crossplatform Manifest build.

Reply:
ok, let's tackle that separately https://github.com/hoprnet/hoprnet/issues/5890